mindmeld/models/query_features.py

```diff
@@ -650,39 +672,30 @@ def _extractor(query, resources):
                     # We never want to differentiate between number tokens.
                     # We may need to convert number words too, like "eighty".
                     token = tokens[index]
-                    tok = mask_numerics(token)
+                    tok = re.sub('\d','0',token)
                     ngram.append(tok)
 
                     if kwargs.get(ENABLE_STEMMING, False):
-                        tok_stemmed = mask_numerics(stemmed_tokens[index])
+                        tok_stemmed = re.sub('\d','0',stemmed_tokens[index])
                         stemmed_ngram.append(tok_stemmed)
 
-                freq = resources[WORD_NGRAM_FREQ_RSC].get(" ".join(ngram), 1)
+                freq = resources[WORD_NGRAM_FREQ_RSC].get(" ".join(ngram), 0)
 
-                if freq > threshold:
-                    joined_ngram = " ".join(ngram)
+                joined_ngram = " ".join(ngram) if freq >= threshold else "OOV"
+                ngram_counter.update(
+                    [
+                        "bag_of_words|length:{}|ngram:{}".format(len(ngram), joined_ngram)
+                    ]
+                )
+                if kwargs.get(ENABLE_STEMMING, False):
+                    freq = resources[WORD_NGRAM_FREQ_RSC].get(" ".join(stemmed_ngram), 0)
```

Comment:
Lets try to reuse vars when we can like `" ".join(stemmed_ngram` is used twice here.